# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerUtils.h
# Commit: 185488c90cb8
# Full Hash: 185488c90cb8d3cd7f7041bc457f5e5cbb1617b2
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2021-06-10 15:47:45
# Regressor Bug: 1564235
# File Overlap Count: 2
# Description:
#   Bug 1564235 - P3 Create IPC for NavigationPreload in PServiceWorkerRegistration.ipdl. r=dom-worker-reviewers,asuth
#   
#   This patch implements the IPC for NavigationPreload APIs.
#   The IPC methods are created on PServiceWorkerRegistration.
#   
# ==============================================================================

diff -r afae3c50f7ac -r 185488c90cb8 dom/serviceworkers/ServiceWorkerUtils.h
--- a/dom/serviceworkers/ServiceWorkerUtils.h	Thu Jun 10 06:01:26 2021 +0000
+++ b/dom/serviceworkers/ServiceWorkerUtils.h	Thu Jun 10 06:01:27 2021 +0000
@@ -7,6 +7,7 @@
 #define _mozilla_dom_ServiceWorkerUtils_h
 
 #include "mozilla/MozPromise.h"
+#include "mozilla/dom/IPCNavigationPreloadState.h"
 #include "mozilla/dom/ServiceWorkerRegistrationDescriptor.h"
 #include "nsTArray.h"
 
@@ -22,6 +23,7 @@
 class ClientInfo;
 class ServiceWorkerRegistrationData;
 class ServiceWorkerRegistrationDescriptor;
+struct NavigationPreloadState;
 
 typedef MozPromise<ServiceWorkerRegistrationDescriptor, CopyableErrorResult,
                    false>
@@ -42,6 +44,9 @@
 
 typedef std::function<void(ErrorResult&&)> ServiceWorkerFailureCallback;
 
+typedef std::function<void(NavigationPreloadState&&)>
+    NavigationPreloadGetStateCallback;
+
 bool ServiceWorkerRegistrationDataIsValid(
     const ServiceWorkerRegistrationData& aData);
 